vm/chef/cookbooks/opencart/files/opencart-utils

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -eu
+#
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source /opt/c2d/c2d-utils || exit 1
+
+readonly opencart_folder="/var/www/html/store"
+
+#######################################
+# Checks if a host and port are opened
+# Arguments:
+#   Host name
+#   Port number
+#######################################
+function wait_for_port() {
+  local HOST="$1"
+  local PORT="$2"
+  timeout --preserve-status 300 bash -c "until echo > /dev/tcp/${HOST}/${PORT}; do sleep 2; done"
+  if [[ "$?" -ne 0 ]]; then
+    exit 1
+  fi
+}
+
+# Setup MySQL user
+function setup_mysql_user() {
+  local -r db_user="$1"
+  local -r db_password="$2"
+  local -r db_name="$3"
+
+  mysql -u root -e "CREATE USER '${db_user}'@'localhost' IDENTIFIED BY '${db_password}';"
+  mysql -u root -e "GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, INDEX, ALTER, CREATE TEMPORARY TABLES ON ${db_name}.* TO '${db_user}'@'localhost' IDENTIFIED BY '${db_password}';";
+  mysql -u root -e "FLUSH PRIVILEGES"
+}
+
+# Prepare files to installer
+function setup_opencart_installer() {
+  local -r src_installer_folder="/usr/src/opencart"
+
+  cp -rf "${src_installer_folder}/upload" "${opencart_folder}"
+  cp -f "${src_installer_folder}/composer.json" "${opencart_folder}/composer.json"
+
+  cd "${opencart_folder}"
+  composer install --no-interaction
+  rm -rf system/storage/vendor
+  composer update --no-interaction
+}
+
+function install_opencart_db() {
+  local -r db_user="$1"
+  local -r db_password="$2"
+  local -r db_name="$3"
+  local -r admin_user="$4"
+  local -r admin_password="$5"
+  local -r email="$6"
+  local -r url="$7"
+
+  cd "${opencart_folder}"
+  php install/cli_install.php install \
+    --db_hostname localhost \
+    --db_username "${db_user}" \
+    --db_password "${db_password}" \
+    --db_database "${db_name}" \
+    --db_driver mysqli \
+    --db_port 3306 \
+    --username "${admin_user}" \
+    --password "${admin_password}" \
+    --email "${email}" \
+    --http_server "${url}"
+}
+
+function patch_file() {
+  local -r contents="$1"
+  local -r target_file="$2"
+  local -r target_line="$3"
+
+  sed -i "${target_line}i${contents}" "${target_file}"
+}
+
+function enable_apache_config() {
```

Comment:
This function is not readable, maybe better replace with ready config or patch with diff file etc.?

Reply:
👍🏻 Sure that.

Reply:
Moved to imagebuilding process. A resource was added to apache2 cookbook and reused in opencart recipe.